recipes-graphics/spark/pxcore-libnode.inc

```diff
@@ -0,0 +1,137 @@
+include pxcore.inc
+
+#DEPENDS += " python-native"
+DEPENDS += " qemu-native "
+
+COMPATIBLE_MACHINE_armv4 = "(!.*armv4).*"
+COMPATIBLE_MACHINE_armv5 = "(!.*armv5).*"
+COMPATIBLE_MACHINE_mips64 = "(!.*mips64).*"
+
+inherit cmake pythonnative
+
+# v8 errors out if you have set CCACHE
+CCACHE = ""
+
+def map_nodejs_arch(a, d):
+    import re
+
+    if   re.match('i.86$', a): return 'ia32'
+    elif re.match('x86_64$', a): return 'x64'
+    elif re.match('aarch64$', a): return 'arm64'
+    elif re.match('powerpc64$', a): return 'ppc64'
+    elif re.match('powerpc$', a): return 'ppc'
+    return a
+
+ARCHFLAGS_arm = "${@bb.utils.contains('TUNE_FEATURES', 'callconvention-hard', '--with-arm-float-abi=hard', '--with-arm-float-abi=softfp', d)} \
+                 ${@bb.utils.contains('TUNE_FEATURES', 'neon', '--with-arm-fpu=neon', \
+                    bb.utils.contains('TUNE_FEATURES', 'vfpv3d16', '--with-arm-fpu=vfpv3-d16', \
+                    bb.utils.contains('TUNE_FEATURES', 'vfpv3', '--with-arm-fpu=vfpv3', \
+                    '--with-arm-fpu=vfp', d), d), d)}"
+GYP_DEFINES_append_mipsel = " mips_arch_variant='r1' "
+ARCHFLAGS ?= ""
+
+# V8's JIT infrastructure requires binaries such as mksnapshot and
+# mkpeephole to be run in the host during the build. However, these
+# binaries must have the same bit-width as the target (e.g. a x86_64
+# host targeting ARMv6 needs to produce a 32-bit binary). Instead of
+# depending on a third Yocto toolchain, we just build those binaries
+# for the target and run them on the host with QEMU.
+python do_create_v8_qemu_wrapper () {
+    """Creates a small wrapper that invokes QEMU to run some target V8 binaries
+    on the host."""
+
+    qemu_binary=qemu_target_binary(d)
+    if qemu_binary == "qemu-allarch":
+        qemu_binary = "qemuwrapper"
+
+    qemu_options = d.getVar("QEMU_OPTIONS", True)
+
+    rootfs_path=d.getVar('STAGING_DIR_HOST', True)
+    libdir_qemu=d.expand('${STAGING_DIR_HOST}${libdir}')
+    base_libdir_qemu=d.expand('${STAGING_DIR_HOST}${base_libdir}')
+
+    qemu_cmd= qemu_binary + " " + qemu_options + " -L " + rootfs_path \
+            + " -E LD_LIBRARY_PATH=" + libdir_qemu + ":"+ base_libdir_qemu + " "
+
+    wrapper_path = d.expand('${S}/examples/pxScene2d/external/v8-qemu-wrapper.sh')
+    with open(wrapper_path, 'w') as wrapper_file:
+        wrapper_file.write("""#!/bin/sh
+
+# This file has been generated automatically.
+# It invokes QEMU to run binaries built for the target in the host during the
+# build process.
+%s "$@"
```

Comment:
Shouldn't this be indented?

Reply:
it is code used to generate file examples/pxScene2d/external/v8-qemu-wrapper.sh file, so it intended how does the data added to this file